DeprecationWarning invalid escape sequence in math docstring

**Describe the bug**
In some docstrings that contain math, an invalid escape sequence is used that causes `pytest` to raise a warning. This could be resolved by "double all backslashes, or use Python raw strings (r"raw")", see [sphinx-doc](https://www.sphinx-doc.org/en/master/usage/restructuredtext/directives.html#math) or [stackoverflow](https://stackoverflow.com/questions/16468397/math-expression-in-docstring-is-not-rendered-correctly-by-sphinx).

**To Reproduce**
Run `pytest`.

**Expected behavior**
No warnings.

**Output**
```
pytest
========================================================================================================================== test session starts ==========================================================================================================================
platform linux -- Python 3.10.12, pytest-7.3.1, pluggy-1.0.0
rootdir: /workspaces/cvxpy
configfile: pyproject.toml
testpaths: cvxpy/tests/
collected 1360 items

...

=========================================================================================================================== warnings summary ============================================================================================================================
cvxpy/reductions/solvers/conic_solvers/sdpa_conif.py:136
  /workspaces/cvxpy/cvxpy/reductions/solvers/conic_solvers/sdpa_conif.py:136: DeprecationWarning: invalid escape sequence '\i'
    """

cvxpy/reductions/cone2cone/soc2psd.py:49
  /workspaces/cvxpy/cvxpy/reductions/cone2cone/soc2psd.py:49: DeprecationWarning: invalid escape sequence '\l'
    """

cvxpy/tests/solver_test_helpers.py:157
  /workspaces/cvxpy/cvxpy/tests/solver_test_helpers.py:157: DeprecationWarning: invalid escape sequence '\i'
    """The convention that we follow for construting the Lagrangian is: 1) Move all

cvxpy/tests/solver_test_helpers.py:194
  /workspaces/cvxpy/cvxpy/tests/solver_test_helpers.py:194: DeprecationWarning: invalid escape sequence '\i'
    """The dual cone to the set of symmetric matrices is the
```

**Version**
 - OS: devcontainer
 - CVXPY Version: master ([3c8ba00](https://github.com/cvxpy/cvxpy/commit/3c8ba004073446a535076b87a42367c3d636780e))

This is now a `SyntaxError` in python 3.11 and can sometimes cause test collection failures

```
ImportError while loading conftest 'our/code/here/conftest.py'.
our/code/here/conftest.py:3: in <module>
    import cvxpy
lib/python3.11/site-packages/cvxpy/__init__.py:57: in <module>
    from cvxpy.problems.problem import Problem as Problem
lib/python3.11/site-packages/cvxpy/problems/problem.py:38: in <module>
    from cvxpy.reductions import InverseData
lib/python3.11/site-packages/cvxpy/reductions/__init__.py:24: in <module>
    from cvxpy.reductions.dcp2cone.dcp2cone import Dcp2Cone
lib/python3.11/site-packages/cvxpy/reductions/dcp2cone/dcp2cone.py:24: in <module>
    from cvxpy.reductions.dcp2cone.canonicalizers import CANON_METHODS as cone_canon_methods
lib/python3.11/site-packages/cvxpy/reductions/dcp2cone/canonicalizers/__init__.py:44: in <module>
    from cvxpy.reductions.dcp2cone.canonicalizers.suppfunc_canon import (
lib/python3.11/site-packages/cvxpy/reductions/dcp2cone/canonicalizers/suppfunc_canon.py:5: in <module>
    from cvxpy.reductions.solvers.conic_solvers.scs_conif import (
lib/python3.11/site-packages/cvxpy/reductions/solvers/conic_solvers/__init__.py:34: in <module>
    from .sdpa_conif import SDPA
E     File "lib/python3.11/site-packages/cvxpy/reductions/solvers/conic_solvers/sdpa_conif.py", line 136
E       """
E       ^^^
E   SyntaxError: invalid escape sequence '\i'
```

--- a/cvxpy/reductions/cone2cone/soc2psd.py
+++ b/cvxpy/reductions/cone2cone/soc2psd.py
@@ -47,7 +47,7 @@ class SOC2PSD(Reduction):
 
         for constraint in soc_constraints:
             """
-            The SOC constraint :math:`\\lVert X \\rVert_2 \leq t` is modeled by `t` and `X`.
+            The SOC constraint :math:`\\lVert X \\rVert_2 \\leq t` is modeled by `t` and `X`.
             We extract these `t` and `X` from the SOC constraint object.
             """
             t, X = constraint.args

--- a/cvxpy/reductions/cone2cone/soc2psd.py
+++ b/cvxpy/reductions/cone2cone/soc2psd.py
@@ -46,8 +46,8 @@ class SOC2PSD(Reduction):
         soc_id_from_psd = {}
 
         for constraint in soc_constraints:
-            """
-            The SOC constraint :math:`\\lVert X \\rVert_2 \\leq t` is modeled by `t` and `X`.
+            r"""
+            The SOC constraint :math:`\lVert X \rVert_2 \leq t` is modeled by `t` and `X`.
             We extract these `t` and `X` from the SOC constraint object.
             """
             t, X = constraint.args

--- a/cvxpy/reductions/solvers/conic_solvers/sdpa_conif.py
+++ b/cvxpy/reductions/solvers/conic_solvers/sdpa_conif.py
@@ -133,7 +133,7 @@ class SDPA(ConicSolver):
             return failure_solution(status)
 
     def solve_via_data(self, data, warm_start: bool, verbose: bool, solver_opts, solver_cache=None):
-        """
+        r"""
         CVXPY represents cone programs as
             (P) min_x { c^T x : A x + b \in K } + d
 

--- a/cvxpy/tests/solver_test_helpers.py
+++ b/cvxpy/tests/solver_test_helpers.py
@@ -154,7 +154,7 @@ class SolverTestHelper:
             self.tester.fail(msg)
         bad_norms = []
 
-        """The convention that we follow for construting the Lagrangian is: 1) Move all
+        r"""The convention that we follow for construting the Lagrangian is: 1) Move all
         explicitly passed constraints to the problem (via Problem.constraints) into the
         Lagrangian --- dLdX == 0 for any such variables 2) Constraints that have
         implicitly been imposed on variables at the time of declaration via specific
@@ -191,7 +191,7 @@ class SolverTestHelper:
                     if diag_entries > 10**(-places):
                         bad_norms.append((dual_cone_violation, opt_var))
                 elif opt_var.is_symmetric():
-                    """The dual cone to the set of symmetric matrices is the
+                    r"""The dual cone to the set of symmetric matrices is the
                     set of skew-symmetric matrices, so we check if dLdX \in
                     set(skew-symmetric-matrices)
                     g[opt_var] is the problematic gradient in question"""

--- a/pyproject.toml
+++ b/pyproject.toml
@@ -4,7 +4,7 @@ select = [
     "F",
     "I",
     "NPY201",
-    "W605",  # Check for invalid escape sequences in docstrings (errors in py > 3.11)
+    "W605",  # Check for invalid escape sequences in docstrings (errors in py >= 3.11)
 ]
 line-length = 100
 exclude = [

--- a/pyproject.toml
+++ b/pyproject.toml
@@ -1,5 +1,11 @@
 [tool.ruff]
-select = ["E", "F", "I", "NPY201"]
+select = [
+    "E",
+    "F",
+    "I",
+    "NPY201",
+    "W605",  # Check for invalid escape sequences in docstrings (errors in py > 3.11)
+]
 line-length = 100
 exclude = [
     "build",

